fail2ban/server/filter.py

```diff
@@ -354,34 +375,22 @@ def getIgnoreIP(self):
 	#
 	# Check if the given IP address matches an IP address/DNS or a CIDR
 	# mask in the ignore list.
-	# @param ip IP address
+	# @param ip IP address object
 	# @return True if IP address is in ignore list
 
 	def inIgnoreIPList(self, ip, log_ignore=False):
```

Comment:
looking at all the tests needing wrapping of `ip:str -> IpAddr(ip)`, may be worth supporting here both str and IpAddr specification with a quick check or even making a "copy constructor" form `IpAddr(ipaddr)` or `asanyipaddr` (numpy rotten soul speaking as for analog to `asanyarray`) helper constructor which would simply assure having an IpAddr object out of string or passing along original IpAddr object if that one given?  That would remove majority of necessary tune ups needed for tests